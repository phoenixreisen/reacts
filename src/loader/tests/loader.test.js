import renderer from 'react-test-renderer';
import Loader from '../loader.r';
import React from "react";

test('Loader should have some default classes & default text + icon', () => {
    const component = renderer.create(<Loader />);
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
});

test('Loader should show customized text & icon', () => {
    const component = renderer.create(
        <Loader text="WORK HARD, PLAY HARD" iconname="fas fa-blabla" />
    );
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
});

test('Loader should set an overlay class', () => {
    const component = renderer.create(
        <Loader text="ICH BIN EIN OVERLAY" iconname="fa-blublu" type="overlay" />
    );
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
});