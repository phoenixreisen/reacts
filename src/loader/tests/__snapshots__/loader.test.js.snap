// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should have some default classes & default text + icon 1`] = `
<article
  className="loader"
>
  <p
    className="loader__spinner"
  >
    <i
      className="fab fa-cuttlefish fa-spin"
    />
  </p>
  <p
    className="loader__text"
  >
    Daten werden geladen...
  </p>
</article>
`;

exports[`Loader should set an overlay class 1`] = `
<article
  className="loader loader--overlay"
>
  <p
    className="loader__spinner"
  >
    <i
      className="fa-blublu fa-spin"
    />
    <br />
    ICH BIN EIN OVERLAY
  </p>
</article>
`;

exports[`Loader should show customized text & icon 1`] = `
<article
  className="loader"
>
  <p
    className="loader__spinner"
  >
    <i
      className="fas fa-blabla fa-spin"
    />
  </p>
  <p
    className="loader__text"
  >
    WORK HARD, PLAY HARD
  </p>
</article>
`;
